Used cmake-format on CMakeLists.txt

BujSet opened the pull request:
### Summary
`Fixes #<10736>`  : https://github.com/pytorch/executorch/issues/10736#event-17545114889

### Changes I Made

`cmake-format -i CMakeLists.txt`

`cmake-lint CMakeLists.txt`

The linting reported an [E1122] Duplicate keyword argument error for variable DESTINATION. I deleted the earlier setting of this variable, as the comment seemed to imply that `lib` should *not* be the appropriate destination.

### Test plan
Successfully ran the commands in the [building from source](https://github.com/BujSet/executorch/blob/main/docs/source/using-executorch-building-from-source.md) guide. I did try running the test commands from the [Testing](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#testing), however, these reported error on a clean install. No new errors were reported when I made these changes.

BujSet committed:
Used cmake-format on CMakeLists.txt

BujSet commented:
@pytorchbot label "release notes: none"

jathu commented on CMakeLists.txt:
```diff
@@ -45,7 +45,8 @@
 
-# MARK: - Start EXECUTORCH_H12025_BUILD_MIGRATION --------------------------------------------------
+# MARK: - Start EXECUTORCH_H12025_BUILD_MIGRATION
+# --------------------------------------------------
```
```suggestion
```

jathu commented on CMakeLists.txt:
```diff
@@ -82,24 +83,25 @@ include(${PROJECT_SOURCE_DIR}/tools/cmake/preset/default.cmake)
 
-# MARK: - End EXECUTORCH_H12025_BUILD_MIGRATION ----------------------------------------------------
+# MARK: - End EXECUTORCH_H12025_BUILD_MIGRATION
+# ----------------------------------------------------
```
```suggestion
```

jathu commented on CMakeLists.txt:
```diff
@@ -82,24 +83,25 @@ include(${PROJECT_SOURCE_DIR}/tools/cmake/preset/default.cmake)
-# but this rpath is not automatically added because at build time it seems `portable_lib`
-# is being built under the same directory, so no extra rpath is being added. To
-# properly fix this we need to install `portable_lib` into the correct path.
+# the rpath (per library?) TODO: Doesn't work for us right now because we are
+# not installing .so's into the correct locations. For example we have
```
I think we should keep the TODO as a separate paragraph like it was before

BujSet committed:
Updates to comments

Removing unnecessary comment lines used for segmenting the file and
splitting larger comments as they were before the cmake-format pass.
